src/ripple/app/ledger/InboundLedgers.cpp

```diff
@@ -67,26 +73,29 @@ class InboundLedgersImp
                 auto it = mLedgers.find (hash);
                 if (it != mLedgers.end ())
                 {
-                    // Don't touch failed acquires so they can expire
-                    if (! it->second->isFailed ())
+                    inbound = it->second;
+
+                    if (! inbound->isFailed ())
                     {
-                        it->second->update (seq);
-                        if (it->second->isComplete ())
-                            ret = it->second->getLedger ();
+                        // Don't touch failed acquires so they can expire
+                        inbound->update (seq);
                     }
-
                 }
                 else
                 {
-                    auto il = std::make_shared <InboundLedger> (hash, seq, reason, std::ref (m_clock));
-                    mLedgers.insert (std::make_pair (hash, il));
-                    il->init (sl);
+                    inbound = std::make_shared <InboundLedger> (hash, seq,
+                        reason, std::ref (m_clock));
+                    mLedgers.insert (std::make_pair (hash, inbound));
```

Comment:
`mLedgers.insert ({hash, inbound});` !

This is an idiom I'm seeing in people's new code dealing with `std::map` and `std::unordered_map`.